spacetemplate/repository.go

```diff
@@ -0,0 +1,286 @@
+package spacetemplate
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/space"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/almighty/almighty-core/workitem/link"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/net/context"
+)
+
+// Repository describes interactions with space templates
+type Repository interface {
+	Create(ctx context.Context, st *SpaceTemplate) error
+	List(ctx context.Context) ([]SpaceTemplate, error)
+	Load(ctx context.Context, spaceTemplateID uuid.UUID) (*SpaceTemplate, error)
+	// Apply tries to load the space template id as stored in the space and then
+	// creates all the artifacts like work item type and work item link types
+	Apply(ctx context.Context, spaceTemplateID uuid.UUID, spaceID uuid.UUID) error
```

Comment:
why do you need to pass the `spaceTemplateID` here if it's already stored in the `space` itself ?